[tp] add clarification to doc and improve TP examples (#117618)
pytorch/pytorch: docs/source/distributed.tensor.parallel.rst

  :members:
  :undoc-members:

.. note:: when using the ``Shard(dim)`` as the input/output layouts for the above

>   ParallelStyles, we assume the input/output activation tensors are evenly sharded on

Reviewer: nit: 
```suggestion
  ``ParallelStyle`` s, we assume the input/output activation tensors are evenly sharded on
```

>   the tensor dimension ``dim`` on the DeviceMesh that TP operates on. For instance,

Reviewer: nit:
```suggestion
  the tensor dimension ``dim`` on the ``DeviceMesh`` that TP operates on. For instance,
```

>   ``RowwiseParallel`` accepts input that is sharded on the last dimension, it assumes

Reviewer: nit:
```suggestion
  since ``RowwiseParallel`` accepts input that is sharded on the last dimension, it assumes
```

>   the input tensor already been evenly sharded on the last dimension.

Author: @awgu I added some clarification about uneven sharding in the doc

>   the input tensor already been evenly sharded on the last dimension.

Reviewer: nit:
```suggestion
  the input tensor has already been evenly sharded on the last dimension.
```

  :members:
  :undoc-members:

.. note:: when using the ``Shard(dim)`` as the input/output layouts for the above
  ``ParallelStyle``s, we assume the input/output activation tensors are evenly sharded on
  the tensor dimension ``dim`` on the ``DeviceMesh`` that TP operates on. For instance,
  since ``RowwiseParallel`` accepts input that is sharded on the last dimension, it assumes

>   the input tensor has already been evenly sharded on the last dimension. If needs uneven

Reviewer 2: nit- if needs
